chitterView.test.js

```diff
@@ -0,0 +1,214 @@
+/**
+ * @jest-environment jsdom
+ */
+
+const fs = require("fs");
+const { default: JSDOMEnvironment } = require("jest-environment-jsdom");
+const ChitterModel = require("./chitterModel");
+const ChitterView = require("./chitterView");
+const DisplaySinglePeep = require("./displaySinglePeep");
+const testPostPeepResponse = require("./testPostPeepResponse");
+let view,
+  model,
+  mockApi,
+  displaySinglePeep,
+  testPeepsArray,
+  testUserSessionResponse,
+  fakeCallbackResponse;
+
+describe("ChitterView", () => {
+  beforeEach(() => {
+    document.body.innerHTML = fs.readFileSync("./index.html");
+    model = new ChitterModel();
+    testPeepsArray = require("./testPeepsArray");
+    testUserSessionResponse = require("./testUserSessionResponse");
+    fakeCallbackResponse = jest.fn();
+    mockApi = {
+      fetchPeeps: jest.fn((callback) => callback(fakeCallbackResponse())),
+      createUser: jest.fn((username, password, callback) =>
+        callback(fakeCallbackResponse())
+      ),
+      loginUser: jest.fn((username, password, callback) =>
+        callback(fakeCallbackResponse())
+      ),
+      postPeep: jest.fn((sessionKey, userId, peepBody, callback) =>
+        callback(fakeCallbackResponse())
+      ),
+      deletePeep: jest.fn((peepId, sessionKey, callback) =>
+        callback(fakeCallbackResponse())
+      ),
+    };
+
+    displaySinglePeep = new DisplaySinglePeep();
+    view = new ChitterView(model, mockApi, displaySinglePeep);
+  });
+
+  it("displays all Peeps", () => {
+    fakeCallbackResponse.mockReturnValueOnce(testPeepsArray);
+    view.displayPeepsFromApi();
+    expect(mockApi.fetchPeeps).toHaveBeenCalled();
+    const peepDivEls = document.querySelectorAll("div.peep");
+    expect(peepDivEls.length).toBe(2);
+    expect(peepDivEls[0].querySelector(".peep-body").textContent).toBe(
+      "First peep"
+    );
+    expect(peepDivEls[0].querySelector(".peep-user-handle").textContent).toBe(
+      "jony144"
+    );
+    expect(
+      peepDivEls[0].querySelector(".peep-datetime-created").textContent
+    ).toBe("12:33 Sat Aug 20 2022");
+    expect(peepDivEls[0].querySelector(".peep-likes-count").textContent).toBe(
+      "1 like"
+    );
+    expect(peepDivEls[1].querySelector(".peep-body").textContent).toBe(
+      "i'm tiz"
+    );
+    expect(peepDivEls[1].querySelector(".peep-user-handle").textContent).toBe(
+      "tiz"
+    );
+    expect(
+      peepDivEls[1].querySelector(".peep-datetime-created").textContent
+    ).toBe("13:04 Sun Aug 07 2022");
+    expect(peepDivEls[1].querySelector(".peep-likes-count").textContent).toBe(
+      "0 likes"
```

Comment:
The test is confirming you fetched two test peeps. You could get away with testing all elements on the first peep and then just one element to confirm the second peep is there. This simplifies the test without a big compromise in integrity.